File: 2026/03- March/27- Friday/solution.js
//updated 
class Solution {
    areSimilar(mat, k) {
        const m = mat.length;
        const n = mat[0].length;

        const shift = k % n;

        if (shift === 0) return true;

        for (let i = 0; i < m; i++) {
            const row = mat[i];
            let shifted;

            if (i % 2 === 0) {
                // left shift
                shifted = row.slice(shift).concat(row.slice(0, shift));
            } else {
                // right shift
                shifted = row.slice(-shift).concat(row.slice(0, -shift));
            }

            if (shifted.toString() !== row.toString()) {
                return false;
            }
        }

        return true;
    }
}

module.exports = Solution;

File: 2026/03- March/27- Friday/solution.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        int m = mat.length;
        int n = mat[0].length;

        int shift = k % n;

        if (shift == 0) return true;

        for (int i = 0; i < m; i++) {
            int[] row = mat[i];
            int[] shifted = new int[n];

            if (i % 2 == 0) {
                // left shift
                for (int j = 0; j < n; j++) {
                    shifted[j] = row[(j + shift) % n];
                }
            } else {
                // right shift
                for (int j = 0; j < n; j++) {
                    shifted[j] = row[(j - shift + n) % n];
                }
            }

            for (int j = 0; j < n; j++) {
                if (shifted[j] != row[j]) {
                    return false;
                }
            }
        }

        return true;
    }
}
